### PR TITLE
Issue #175: Migration of CommunityController to OpenApi yaml schema 

### LINKED ISSUE #175: Migration of CommunityController to OpenApi yaml schema
This task is a subtask for #169.

The steps that need to be done:
- convert all endpoints from controller to yaml file
- check that endpoints are working, in the same way, like before the change
- check the tests

An example can be found in PR #171.

### PR DESCRIPTION
**Pull request by huyle101**
## 🚀 Description
Added OpenAPI spec for CommunityController

## 📄 Motivation and Context
Closes: #175 

## 🧪 How Has This Been Tested?

- Test cases are passing

- Verified functionalities via running locally Home-Web UI

## 📦 Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## ✅ Checklist

- [x] My code follows the code style of this project(Do your best to follow code styles. If none apply just skip this).
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: service/src/test/java/com/myhome/controllers/CommunityControllerTest.java [hunk 1]
```diff
@@ -159,14 +165,13 @@ void shouldListAllCommunitiesSuccessfully() {
...
+    Set<GetCommunityDetailsResponseCommunity> communityDetailsResponse
         = new HashSet<>();
     communityDetailsResponse.add(
-        new GetCommunityDetailsResponse.Community(
-            COMMUNITY_ID,
-            COMMUNITY_NAME,
-            COMMUNITY_DISTRICT
-        )
+        new GetCommunityDetailsResponseCommunity()
+        .communityId(COMMUNITY_ID)
```
Comment (mslowiak) [inline service/src/test/java/com/myhome/controllers/CommunityControllerTest.java]: Not sure if this formatting is correct.
Can you try to CTR+ALT+L in Intellij?

File: service/src/test/java/com/myhome/controllers/CommunityControllerTest.java [hunk 2]
```diff
@@ -194,19 +199,18 @@ void shouldGetCommunityDetailsSuccessfully() {
...
-        new GetCommunityDetailsResponse.Community(
-            COMMUNITY_ID,
-            COMMUNITY_NAME,
-            COMMUNITY_DISTRICT
-        );
-
-    Set<GetCommunityDetailsResponse.Community> communityDetailsResponse
+    GetCommunityDetailsResponseCommunity communityDetails =
+        new GetCommunityDetailsResponseCommunity()
+        .communityId(COMMUNITY_ID)
```
Comment (mslowiak) [inline service/src/test/java/com/myhome/controllers/CommunityControllerTest.java]: Same as up, formatting

File: service/src/test/java/com/myhome/controllers/CommunityControllerTest.java [hunk 3]
```diff
@@ -255,17 +259,16 @@ void shouldListCommunityAdminsSuccess() {
...
     Set<User> adminsSet = new HashSet<>(admins);
 
-    Set<ListCommunityAdminsResponse.CommunityAdmin> listAdminsResponses = new HashSet<>();
+    Set<ListCommunityAdminsResponseCommunityAdmin> listAdminsResponses = new HashSet<>();
     listAdminsResponses.add(
-        new ListCommunityAdminsResponse.CommunityAdmin(
-            COMMUNITY_ADMIN_ID
-        )
+        new ListCommunityAdminsResponseCommunityAdmin()
+        .adminId(COMMUNITY_ADMIN_ID)
```
Comment (mslowiak) [inline service/src/test/java/com/myhome/controllers/CommunityControllerTest.java]: Formatting

**Review by mslowiak — CHANGES_REQUESTED** [review]

**Comment by huyle101** [comment]
fixed format issue, thanks @mslowiak

**Comment by huyle101** [comment]
I guess the build failed due to the merge recently. (https://github.com/jmprathab/MyHome/runs/1245944712)

**Comment by mslowiak** [comment]
> I guess the build failed due to the merge recently. (https://github.com/jmprathab/MyHome/runs/1245944712)

@huyle1097 yeah, there was a problem with class compilation `WebSecurity`.
Master is fixed for now

**Review by mslowiak — CHANGES_REQUESTED** [review]
Please rebase on the latest master and we can merge this

**Review by mslowiak — APPROVED** [review]

**Comment by huyle101** [comment]
Hi @mslowiak 
Can you put the label  _hacktoberfest-accepted_ for this PR?
Thanks :)

**Comment by mslowiak** [comment]
Sure @huyle1097, however, your pull request should be accepted automatically because our repo has the topic 'hacktoberfest' :)

**Comment by huyle101** [comment]
ah, I see.
Thanks @mslowiak

**Pull request merged**